extension/test/background_test.ts

```diff
@@ -65,26 +87,76 @@ describe('background.ts', function () {
         { config: rcxMain.config }
       );
     });
+
+    it('should throw typeError if tab is undefined', async function () {
+      // }).to.throw(TypeError, 'sender.tab is always defined here.');
+    });
   });
-});
 
-async function sendMessageToBackground({
-  tabId = 0,
-  type,
-  responseCallback = () => {
-    // Do nothing by default.
-  },
-}: {
-  tabId?: number;
-  type: string;
-  responseCallback?: (response: unknown) => void;
-}): Promise<void> {
-  // In background.ts, a promise is passed to `addListener` so we can await it here.
-  // eslint-disable-next-line @typescript-eslint/await-thenable
-  await chrome.runtime.onMessage.addListener.yield(
-    { type: type },
-    { tab: { id: tabId } },
-    responseCallback
-  );
-  return;
-}
+  describe('xsearch', function () {
+    it('should return search results', async function () {
+      const searchStub = sinon.stub().returns(['to eat', 'verb']);
+      rcxMain.search = searchStub;
+      const request = {
+        type: 'xsearch',
+        text: 'た',
+        dictOption: '2',
+      };
+      const sender = { tab: { id: 0 } };
+      const response = sinon.spy();
+
+      // eslint-disable-next-line @typescript-eslint/await-thenable
+      await chrome.runtime.onMessage.addListener.callArgWith(
+        0,
+        request,
+        sender,
+        response
+      );
+
+      expect(searchStub.calledWith(request.text, request.dictOption)).to.be
+        .true;
+
+      expect(response.called).to.be.true;
```

Comment:
I would update this to check that it was called with the right output since it's the key behavior of the code under test.